Kadabra-JS/src-api/kadabra/Utils.ts

```diff
@@ -0,0 +1,210 @@
+    }
+}
+
+/**
+ * Inserts a print statement in the main method of a class.
+ *
+ * @param message - The message to print.
+ */
+export function printOnMain2(message: string): void {
+    for (const $method of Query.search(Class).search(Method, "main")) {
+        beforeExit($method, `System.out.println(${message});`);
+    }
+}
+
+/**
+ * Inserts code before the exit of the main method.
+ *
+ * @param code - The code to insert.
+ */
+export function beforeExitMain(code: string): void {
+    for (const $method of Query.search(Class).search(Method, "main")) {
+        beforeExit($method, code);
+    }
+}
+
+/**
+ * Inserts code before the exit of a method.
+ *
+ * @param method - The method join point.
+ * @param code - The code to insert.
+ */
+export function beforeExit(method: Method, code: string): void {
+    let inserted = false;
+
+    // Try to insert before return statements
+    for (const stmt of Query.searchFrom(method.body, Return)) {
+        stmt.insert("before", code);
+        inserted = true;
+    }
+    if (inserted) return;
+
+    // Try to insert after the last statement (for void methods)
+    for (const stmt of Query.searchFrom(method).search("body.lastStmt")) {
+        stmt.insert("after", code);
+        inserted = true;
+    }
+    if (inserted) return;
+
+    // Else, insert into an empty method
+    for (const body of Query.searchFrom(method).search(Body)) {
+        body.insert("replace", code);
+    }
+}
+
+/**
+ * Generates API names for concurrent package components.
+ *
+ * @returns An object containing the API names.
+ */
+export function getAPINames(): { concurrentPackage: string; channel: string; thread: string; product: string } {
+    return {
+        concurrentPackage: "weaver.kadabra.concurrent",
+        channel: "weaver.kadabra.concurrent.KadabraChannel",
+        thread: "weaver.kadabra.concurrent.KadabraThread",
+        product: "weaver.kadabra.concurrent.Product",
+    };
+}
+
+/**
+ * Generates code to retrieve an integer property from the system.
+ *
+ * @param name - The name of the property.
+ * @param defaultValue - The default value of the property (optional).
+ * @returns The generated code as a string.
+ */
+export function integerProperty(name: any, defaultValue?: string): string {
+    let code = `Integer.parseInt(System.getProperty(${JSON.stringify(name)}`;
+    if (defaultValue !== undefined) {
+        code += `, "${defaultValue}\\"`;
+    }
+    return code + "))";
+}
+
+/**
+ * Retrieves methods based on class and method names.
+ *
+ * @param className - The name of the class.
+ * @param methodName - The name of the method.
+ * @returns An array of method join points.
+ */
+export function getMethod(className: string = ".*", methodName?: string): Method | Array<Method | Method[]> {
+    if (!methodName) {
+        methodName = className;
+        className = ".*";
+    }
+
+    let methods: Method | Array<Method | Method[]> = undefined;
```

Comment:
Re-li agora o código e apercebi-me de que o tipo é este.

```suggestion
let methods: Method | Method[] = undefined;
```

Reply:
Ok, já troquei

Reply:
Já troquei